# Obtención de Estadisticas Descriptivas

En esta actividad trabajaras con el conjunto de datos asignado para el reto. Tienes que replicar los pasos vistos durante las horas de salón en un diferente conjunto de datos.

1. Carga los datos usando tu lector de csv o con pandas. Es recomendable hacerlo con pandas.
2. Verifica la cantidad de datos que tienes, las variables que contiene cada vector de datos e identifica el tipo de variables.
3. Analiza las variables para saber qué representa cada una y en que rangos se encuentran. Si la descripción del problema no te lo indica, utiliza el máximo y el mínimo para encontrarlo.

### Carga de librerias

Estas son las librerias que nos ayudaran al analisis de datos

In [6]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt

### Carga del Archivo: Diamonds are Forever

Con el metodo read_csv de pandas se carga el archivo en forma de Dataframe

In [7]:
diamantes = pd.read_csv("Diamonds are Forever.csv")

## Analisis Inicial

En el analisis se obtienen datos generales del archivo guardado en la variable diamantes. Primero se obtienen sus primeras 5 filas con el metodo head() y sus ultimas 5 filas con el metodo tail()

In [19]:
diamantes.head()

Carat  Colour  Clarity  Lab  Price ($US)
0   0.30       1        5    1        911.4
1   0.30       2        4    1       1057.0
2   0.30       4        2    1       1057.0
3   0.30       4        4    1        882.0
4   0.31       1        4    1       1148.7

In [20]:
diamantes.tail()

Carat  Colour  Clarity  Lab  Price ($US)
303   1.01       6        4    3       5722.5
304   1.02       3        3    3       7557.2
305   1.06       5        3    3       6923.0
306   1.02       5        5    3       6271.3
307   1.09       6        3    3       6374.9

Con el metodo info() podemos observar las variables (columnas) que existen en el dataframe y el tipo de datos que contienen.

In [10]:
diamantes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 308 entries, 0 to 307
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Carat        308 non-null    float64
 1   Colour       308 non-null    int64  
 2   Clarity      308 non-null    int64  
 3   Lab          308 non-null    int64  
 4   Price ($US)  308 non-null    float64
dtypes: float64(2), int64(3)
memory usage: 12.2 KB


In [12]:
diamantes.shape

(308, 5)

Con el metodo describe() se obtienen estadisticas descriptivas del dataframe como lo son el numero de datos en cada columna, su promedio, su desviación estandar, sus valores maximos y minimos, etc.

In [16]:
diamantes.describe()

Carat     Colour     Clarity         Lab   Price ($US)
count  308.000000  308.00000  308.000000  308.000000    308.000000
mean     0.630909    3.75000    3.152597    1.766234   3513.638636
std      0.277183    1.40236    1.293718    0.833283   2382.181001
min      0.180000    1.00000    1.000000    1.000000    446.600000
25%      0.350000    3.00000    2.000000    1.000000   1137.500000
50%      0.620000    4.00000    3.000000    2.000000   2950.500000
75%      0.850000    5.00000    4.000000    3.000000   5212.200000
max      1.100000    6.00000    5.000000    3.000000  11205.600000

In [18]:
diamantes.median()

Carat             0.62
Colour            4.00
Clarity           3.00
Lab               2.00
Price ($US)    2950.50
dtype: float64

### Basándose en la media, mediana y desviación estándar de cada variable, ¿qué conclusiones puedes entregar de los datos?

En base al analisis de datos incial, se puede observar que ninguno de los datos posee valores nulos, con un total de 308 filas y 5 columnas que contienen valores de tipo int64 y float64. Los promedios obtenidos de las variables nos dicen lo siguiente:

- La columna Carat nos indica el peso de los diamantes. El promedio de este es de 0.63 carats, equivalente a 0.13 gramos.
- La columna Colour indica la calidad del color que va de 1=sin color a 6=casi sin color, su promedio 3.75 se encuentra casi a la mitad de este rango.
- Los diamantes tiene un precio promedio de 3513.64 dolares.

La desviacion estandar de cada variable es pequeña, ya que se encuentran entre los valores de 0 a 1. Esto significa que los datos no son tan dispersos y se encuentran cerca del promedio de cada variable.

La media de cada variable nos da una idea de lo que puede ser el promedio, pues los valores mostrados se acercan a los valores de la mediana en la tabla del metodo describe().  
